src/unix-manager.c

```diff
@@ -143,18 +143,27 @@ int UnixNew(UnixCommand * this)
         struct stat stat_buf;
         /* coverity[toctou] */
         if (stat(SOCKET_PATH, &stat_buf) != 0) {
-            SCLogWarning(SC_ERR_INITIALIZATION,
-                    "Unix socket: problem with requested directory in %s: %s",
-                    SOCKET_TARGET, strerror(errno));
-            return 0;
+            int ret;
+            /* coverity[toctou] */
+            ret = mkdir(SOCKET_PATH, S_IRWXU|S_IXGRP|S_IRGRP);
```

Comment:
Here we are doing quite the same as in the previous block code:
https://github.com/inliniac/suricata/pull/2445/commits/c5e550b10d4b61fb5be4b117679bd2cab00420e5#diff-3bf3a0dac6ccf03f336cc9660c505d13R128

I think we should factorize by setting up sockettarget then pass in the code testing for existence of dir and creation if needed.